src/AI-DataFrameInspector/AISpSortTablePresenter.class.st

```diff
@@ -0,0 +1,171 @@
+"
+Class for handling the sort window.
+"
+Class {
+	#name : #AISpSortTablePresenter,
+	#superclass : #AISpDataFramePresenter,
+	#instVars : [
+		'sortType',
+		'columnName',
+		'block',
+		'addButton',
+		'tablePresenter',
+		'applyButton',
+		'cancelButton',
+		'dataRows',
+		'newRow'
+	],
+	#category : #'AI-DataFrameInspector-sub presenters'
+}
+
+{ #category : #'instance creation' }
+AISpSortTablePresenter class >> openOn: aDataFrame [
+
+	^ (self on: aDataFrame) open
+]
+
+{ #category : #adding }
+AISpSortTablePresenter >> addButtonAction [
+
+	newRow := Dictionary new.
+	newRow at: 'columnName' put: columnName selectedItem.
+	newRow at: 'sortType' put: sortType selectedItem.
+	newRow at: 'sortBlock' put: block text.
+	dataRows add: newRow.
+	tablePresenter items: dataRows
+]
+
+{ #category : #action }
+AISpSortTablePresenter >> applyButtonAction [
+
+	| currentRow |
+	currentRow := Dictionary new.
+	(1 to: dataRows size) reverseDo: [ :rowNumber |
+		currentRow := dataRows at: rowNumber.
+		dataFrame
+			sortBy: (currentRow at: 'columnName')
+			using: (BlockClosure readFromString: (currentRow at: 'sortBlock')) ]
```

Comment:
```suggestion
	dataRows valuesDo: [ :currentRow |
		dataFrame
			sortBy: (currentRow at: 'columnName')
			using: (BlockClosure readFromString: (currentRow at: 'sortBlock')) ]
```

I have the impression that this piece of code is the same but more readable

Reply:
dataRows is an ordered collection, it doesn't have #valuesDo: and I also need to do it in reverse order.